src/services/webPlatformUtils.service.ts

```diff
@@ -313,11 +314,13 @@ export class WebPlatformUtilsService implements PlatformUtilsService {
         return false;
     }
 
-    getDefaultSystemTheme() {
-        return Promise.resolve(null as 'light' | 'dark');
+    getDefaultSystemTheme(): Promise<'light' | 'dark'> {
+        return Promise.resolve(this.prefersColorSchemeDark.matches ? 'dark' : 'light');
     }
 
-    onDefaultSystemThemeChange() {
-        /* noop */
+    onDefaultSystemThemeChange(callback: ((theme: 'light' | 'dark') => unknown)) {
+        this.prefersColorSchemeDark.addListener(({ matches }) => {
```

Comment:
I don't know much about media queries, but `addListener` appears to be deprecated: https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList/addListener. Is there a more future-proof way you can do this?

Reply:
Sorted, I'll do a PR for fixing this in the browser too

Reply:
PR for the browser: [#1898](https://github.com/bitwarden/browser/pull/1898)